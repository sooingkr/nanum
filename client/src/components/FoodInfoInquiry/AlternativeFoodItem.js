import React from 'react';
import { PropTypes } from 'prop-types';

import { Link } from 'react-router-dom';
import { Image } from 'react-bootstrap';


const AlternativeFoodItem = ({ food: { imgSrc, foodName, foodInfo, foodId } }) => {

  return (
    <div className='food-slider__food'>
      <Link to={{ pathname: "/product/id/" + foodId }}>
        <Image responsive src={imgSrc} alt={foodName}></Image>
      </Link>
      <div className="food-slider__info">{foodInfo}</div>
    </div>
  )
}

AlternativeFoodItem.propTypes = {
  food: PropTypes.shape({
    foodId: PropTypes.oneOfType([
      PropTypes.number,
      PropTypes.string
    ]).isRequired,
    imgSrc: PropTypes.string.isRequired,
    foodName: PropTypes.string.isRequired,
    foodInfo: PropTypes.string.isRequired
  })
}

export default AlternativeFoodItem;