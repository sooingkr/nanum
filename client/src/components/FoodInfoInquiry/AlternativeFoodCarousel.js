import React from 'react';
import { PropTypes } from 'prop-types';

import { isArray } from 'lodash';

import Carousel from '../Common/Carousel';
import AlternativeFoodItem from './AlternativeFoodItem';

const AlternativeFoodCarousel = ({foods}) => {

    const foodCarouselSettings = {
      arrows: true,
      dots: false,
      infinite: true,
      speed: 500,
      slidesToShow: 2,
      slidesToScroll: 2,
      responsive: [
        { breakpoint: 1025, settings: { slidesToShow: 1, slidesToScroll: 1 }},
        { breakpoint: 769, settings: { slidesToShow: 2, slidesToScroll: 2 }},
        { breakpoint: 480, settings: { slidesToShow: 2, slidesToScroll: 2, vertical: true}}]
    };

  return (
      <Carousel className='food-carousel__sliders' settings={foodCarouselSettings}>
        {
          isArray(foods) && foods.map((food, index) =>
            <div className="food-carousel__item" key={ index }>
              <AlternativeFoodItem food={food}/>
            </div>
          )
        }
      </Carousel>
    );
}

AlternativeFoodCarousel.propTypes = {
  foods: PropTypes.array
}

export default AlternativeFoodCarousel;