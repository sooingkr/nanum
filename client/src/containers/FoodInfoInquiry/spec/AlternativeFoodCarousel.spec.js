import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import AlternativeFoodCarousel from "../../../components/FoodInfoInquiry/AlternativeFoodCarousel";
import AlternativeFoodItem from "../../../components/FoodInfoInquiry/AlternativeFoodItem";


describe('AlternativeFoodCarousel component', () => {
  const wrapper = shallow(<AlternativeFoodCarousel/>);

  it('should render <AlternativeFoodCarousel /> components', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render div className="food-carousel__sliders"', () => {
    expect(wrapper.find(".food-carousel__sliders").length).toBe(1);
  });

  it('should render div className="food-food-carousel__item"', () => {
    expect(wrapper.find(".food-food-carousel__item").length).toBeGreaterThanOrEqual(1);
  });

  const FoodItemCarousel = wrapper.find(AlternativeFoodItem);

  describe('AlternativeFoodItem component with props', () => {
    it('should render <AlternativeFoodItem props/>', () => {
      expect(FoodItemCarousel.prop('food').length).toBeGreaterThanOrEqual(1);
    });

    it('should render div className="food-slider__food"', () => {
      expect(FoodItemCarousel.find(".food-slider__food").length).toBeGreaterThanOrEqual(1);
    });

    it('should render div className="food-slider__info"', () => {
      expect(FoodItemCarousel.find(".food-slider__info").length).toBeGreaterThanOrEqual(1);
    });
  });
});