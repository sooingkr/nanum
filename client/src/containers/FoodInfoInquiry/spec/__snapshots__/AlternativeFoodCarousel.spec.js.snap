// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlternativeFoodCarousel component should render <AlternativeFoodCarousel /> components 1`] = `
<Carousel
  className="food-carousel__sliders"
  settings={
    Object {
      "arrows": true,
      "dots": false,
      "infinite": true,
      "responsive": Array [
        Object {
          "breakpoint": 1024,
          "settings": Object {
            "slidesToScroll": 1,
            "slidesToShow": 1,
          },
        },
        Object {
          "breakpoint": 480,
          "settings": Object {
            "slidesToScroll": 1,
            "slidesToShow": 1,
          },
        },
      ],
      "slidesToScroll": 2,
      "slidesToShow": 2,
      "speed": 500,
    }
  }
>
  <div
    className="food-carousel__item"
    key="0"
  >
    <AlternativeFoodItem
      food={
        Object {
          "foodId": "1",
          "foodInfo": "철분과 비타민A 가 함유된
 HACCP 인증 식품인 요거트 그릭을
 대체식품으로 추천합니다.",
          "foodName": "캔모아 딸기 아이스크림",
          "imgSrc": "http://data.whicdn.com/images/25342540/original.jpg",
        }
      }
    />
  </div>
  <div
    className="food-carousel__item"
    key="1"
  >
    <AlternativeFoodItem
      food={
        Object {
          "foodId": "2",
          "foodInfo": "철분과 비타민A 가 함유된
 HACCP 인증 식품인 요거트 그릭을
 대체식품으로 추천합니다.",
          "foodName": "캔모아 딸기 아이스크림",
          "imgSrc": "https://halachaaday.files.wordpress.com/2014/12/546566_food_dessert_yummy_sweet_cheris_chocolate_plate_2500x1923_www-gdefon-ru.jpg",
        }
      }
    />
  </div>
  <div
    className="food-carousel__item"
    key="2"
  >
    <AlternativeFoodItem
      food={
        Object {
          "foodId": "3",
          "foodInfo": "철분과 비타민A 가 함유된
 HACCP 인증 식품인 요거트 그릭을
 대체식품으로 추천합니다.",
          "foodName": "캔모아 딸기 아이스크림",
          "imgSrc": "https://happyspeedy.com/sites/default/files/styles/product_display_node_large/public/straw_shortcake_bar.jpg?itok=IRm7bcEb",
        }
      }
    />
  </div>
  <div
    className="food-carousel__item"
    key="3"
  >
    <AlternativeFoodItem
      food={
        Object {
          "foodId": "4",
          "foodInfo": "철분과 비타민A 가 함유된
 HACCP 인증 식품인 요거트 그릭을
 대체식품으로 추천합니다.",
          "foodName": "캔모아 딸기 아이스크림",
          "imgSrc": "https://happyspeedy.com/sites/default/files/styles/product_display_node_large/public/straw_shortcake_bar.jpg?itok=IRm7bcEb",
        }
      }
    />
  </div>
  <div
    className="food-carousel__item"
    key="4"
  >
    <AlternativeFoodItem
      food={
        Object {
          "foodId": "5",
          "foodInfo": "철분과 비타민A 가 함유된
 HACCP 인증 식품인 요거트 그릭을
 대체식품으로 추천합니다.",
          "foodName": "캔모아 딸기 아이스크림",
          "imgSrc": "https://happyspeedy.com/sites/default/files/styles/product_display_node_large/public/straw_shortcake_bar.jpg?itok=IRm7bcEb",
        }
      }
    />
  </div>
  <div
    className="food-carousel__item"
    key="5"
  >
    <AlternativeFoodItem
      food={
        Object {
          "foodId": "6",
          "foodInfo": "철분과 비타민A 가 함유된
 HACCP 인증 식품인 요거트 그릭을
 대체식품으로 추천합니다.",
          "foodName": "캔모아 딸기 아이스크림",
          "imgSrc": "https://happyspeedy.com/sites/default/files/styles/product_display_node_large/public/straw_shortcake_bar.jpg?itok=IRm7bcEb",
        }
      }
    />
  </div>
</Carousel>
`;
